src/include/migraphx/generic_float.hpp

```diff
@@ -121,13 +135,13 @@ struct __attribute__((packed)) generic_float
                 int shift  = 0;
                 f.mantissa = mantissa;
 
-                while((f.mantissa & (1 << MantissaSize)) == 0)
+                if(MantissaSize < float32_parts::mantissa_width())
                 {
-                    f.mantissa <<= 1;
-                    shift++;
+                    shift = MantissaSize - (32 - countl_zero(mantissa));
```

Comment:
I see, you are wanting to compute the log2 of the mantissa. You should do `sizeof(mantissa)` instead of `32`.

Reply:
Then do `sizeof(unsigned int)`.

Reply:
sizeof(unsigned int) * 8?